lib/node_modules/@stdlib/stats/mskmax/lib/main.js

```diff
@@ -0,0 +1,115 @@
+/**
+* @license Apache-2.0
+*
+* Copyright (c) 2025 The Stdlib Authors.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var dtypes = require( '@stdlib/ndarray/dtypes' );
+var gmskmax = require( '@stdlib/stats/base/ndarray/mskmax' );
+var dmskmax = require( '@stdlib/stats/base/ndarray/dmskmax' );
+var smskmax = require( '@stdlib/stats/base/ndarray/smskmax' );
+var factory = require( '@stdlib/ndarray/base/binary-reduce-strided1d-dispatch-factory' );
+
+
+// VARIABLES //
+
+var idtypes = dtypes( 'real_and_generic' );
+var mdtypes = dtypes( 'integer' );
+var odtypes = dtypes( 'real_and_generic' );
+var policies = {
+	'output': 'promoted',
```

Comment:
This is not correct. We should not be promoting here. A mask array should have no affect on the output dtype.